web/src/hashchange.js

```diff
@@ -487,6 +487,11 @@ function hashchanged(from_reload, e) {
         return undefined;
     }
 
+    // Hide popovers when user is switching tab / section the settings menu
+    if (hash_parser.is_in_specified_hash_category(["stream", "channels", "groups"])) {
+        popovers.hide_all();
```

Comment:
yeah, initially I thought of implementing it that way. But I'm not sure about the reason why (if there was any) `popovers.hide_all()` was put after the `is_overlay_hash` check in the first place, so I opted implementing an allowlist kind of check just to be conservative. 
Though when testing locally I couldn't find any regression when it was only just `popovers.hide_all()` before the overlay check with out the extra `is_in_specified_hash_category` check. For now I'll revert it to just `popovers.hide_all()` since I couldn't think of any breaking regression, so its probably fine to do it that way. :+1:

Reply:
It is good to check if there was any specific reason for the code change by using `git blame`. This line was originally added in 770a7896515b4 and it does not mention any specific reason to call this after the overlay change code.

Anyways, I think it is fine to just hide the popovers in case the view changes, which is what happens on changing the hash, so this change should be fine. We can see if there is any other suggestion/feedback when the PR is reviewed by Tim.